### PR TITLE
UCP: Add non-blocking disconnect.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2329/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1661/ for details (Mellanox internal link).

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -215,18 +215,40 @@ static void ucp_ep_destory_uct_eps(ucp_ep_h ep)
...
+    ucs_debug("disconnect ep %p", ep);
 
     UCS_ASYNC_BLOCK(&worker->async);
     ucp_ep_delete_from_hash(ep);
     ucp_ep_destory_uct_eps(ep);
     UCS_ASYNC_UNBLOCK(&worker->async);
 
     ucs_free(ep);
+
+    return NULL;
```
Comment (alinask) [inline src/ucp/core/ucp_ep.c]: this function should always return NULL?

Comment (yosefe) [inline src/ucp/core/ucp_ep.c]: for now, yes, until i will make it really non-blocking

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2334/ for details.

**Comment by yosefe** [comment]
@shamisp pls review

File: src/ucp/core/ucp_ep.c (hunk 1 above)
Comment (shamisp) [inline src/ucp/core/ucp_ep.c]: please add comment in the code - the code will be extended to actually implement non-blocking disconnect

**Comment by LenaO** [comment]
is the name of the function ucp_diconnect_nb (header-file) or ucp_ep_disconnect_nb (source-code)?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1670/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2339/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1671/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1675/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
👍

**Pull request merged**